Feat/xpub UI

komret opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->
- [x] enable displaying XPUB while device is not connected
- [x] display XPUB on device for confirmation when device is connected
- [x] show warning before displaying in device is not backed up (@mroz22, please check the changes in connect)
- [x] enable closing the no-backup modal for both XPUB and receive address flow and add anchor to the backup link
- [x] unit tests

## Related Issue

Resolve [#3454](https://github.com/trezor/trezor-suite/issues/3454)

## Screenshots:
#### Device not backed up:
![Screenshot 2023-05-31 at 13 22 22](https://github.com/trezor/trezor-suite/assets/42465546/bf90d0e8-6889-4c24-a1d1-4f339f7c0b5f)
#### Confirmation on device:
![Screenshot 2023-05-31 at 15 37 04](https://github.com/trezor/trezor-suite/assets/42465546/8c72b647-4370-435b-bdeb-5fcda403ed4f)
#### Device not connected:
![Screenshot 2023-05-31 at 15 35 43](https://github.com/trezor/trezor-suite/assets/42465546/d56d9957-d8d1-4689-a91a-0c5de092f524)
![Screenshot 2023-05-31 at 15 33 55](https://github.com/trezor/trezor-suite/assets/42465546/d87d2495-8793-4043-8692-ea103128fc86)

dahaca commented on packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx:
```diff
@@ -0,0 +1,117 @@
+                        )}
+                    </CopyButtonWrapper>
+                ) : (
+                    <StyledDeviceDisconnected label={device.label} />
```
`border-radius: 8px;`

Noticed that it's not there on the design but it looks off

dahaca commented on packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx:
```diff
@@ -0,0 +1,95 @@
+            }
+            bottomBar={
+                <>
+                    <Button variant="secondary" onClick={showUnverifiedAndContinue}>
```
nit: `flex: 1` looks nicer

komret replied:
Agreed, but giving it to the primary button instead.

dahaca commented on packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx:
```diff
@@ -0,0 +1,95 @@
+`;
+
+const StyledModal = styled(Modal)`
+    width: 600px;
```
nit: `500px` looks nicer

komret replied:
Done.

dahaca commented on packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx:
```diff
@@ -0,0 +1,95 @@
+import { Button, Image, ModalProps } from '@trezor/components';
+
+const ImageWrapper = styled.div`
+    padding: 60px 0px;
```
nit: `20px 0px` looks nicer

<img width="400" alt="image" src="https://github.com/trezor/trezor-suite/assets/45338719/c029412a-f2c1-4a45-8c8c-122ac311e30e">

komret replied:
Done.

dahaca commented on packages/suite/src/actions/wallet/coinmarket/coinmarketCommonActions.ts:
```diff
@@ -73,7 +73,7 @@ export const verifyAddress =
         const modalPayload = {
             device,
-            address,
+            value: address,
```
Why this change?

komret replied:
It allowed me to spread the payload in [UserContextModal](https://github.com/trezor/trezor-suite/pull/8613/files#diff-ec45a7963475465fb76fbb83c0c32f3368eb57efb9b74ca644f0dfb6f50efcbf), wasn't really necessary...

dahaca commented on packages/suite/src/actions/wallet/publicKeyActions.ts:
```diff
@@ -0,0 +1,82 @@
+import TrezorConnect from '@trezor/connect';
+
+export const openXpubModal =
+    (params?: Partial<Extract<modalActions.UserContextPayload, { type: 'xpub' }>>) =>
```
narrow down the type to `isConfirmed` and  `isCancelable` only maybe?

komret replied:
Done.

dahaca commented on packages/suite/src/actions/wallet/publicKeyActions.ts:
```diff
@@ -0,0 +1,82 @@
+        useEmptyPassphrase: device.useEmptyPassphrase,
+        showOnTrezor: true,
+    };
+    let response;
```
nit: type response as `Success<unknown> | Unsuccessful`?

komret replied:
Done.

dahaca commented on packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx:
```diff
@@ -0,0 +1,117 @@
+                <Value data-test={valueDataTest}>{value}</Value>
+                {device.connected ? (
+                    <CopyButtonWrapper ref={htmlElement}>
+                        {isConfirmed && (
```
makes little sense to not show the copy button if the device is not connected IMO

dahaca replied:
Consulted ondra and he agrees that the copy button can be rendered right away in case the device is not connected.

komret replied:
Agreed, done.

dahaca commented on packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx:
```diff
@@ -0,0 +1,117 @@
+    valueDataTest?: string;
+}
+
+export const ConfirmDeviceScreen = ({
```
Sound a bit like `DevicePromptModal` to me, quite generic. What about `ConfirmValueScreen` or `ConfirmValueOnDevice`?

komret replied:
Renamed to `ConfirmValueOnDevice`.

dahaca commented on packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx:
```diff
@@ -0,0 +1,117 @@
+
+    const dispatch = useDispatch();
+
+    const copy = () => {
```
`document.execCommand` is deprecated, so what about `navigator.clipboard.writeText(text)`? Could be added directly to `copyToClipboard` maybe? @matejkriz  Eliminates the need for a `ref` too.

komret replied:
You're right, refactored!

dahaca replied:
Thank you for spending time on it!

dahaca commented on packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx:
```diff
@@ -0,0 +1,95 @@
+    const dispatch = useDispatch();
+    const { device, isLocked } = useDevice();
+
+    // just to make TS happy
```
we all want it to be happy ^^

komret replied:
I left that comment from a previous version :)

dahaca commented on packages/suite/src/components/suite/modals/confirm/ConfirmUnverified.tsx:
```diff
@@ -0,0 +1,95 @@
+                </>
+            }
+        >
+            <ImageWrapper>
```
nit: add margins to the image instead?

komret replied:
Done.

komret commented:
Added an option to close the modal while confirming on device.

Also moved the warning to the top, it looks better now that there is the copy button:
![Screenshot 2023-06-05 at 19 17 48](https://github.com/trezor/trezor-suite/assets/42465546/5666a01f-dab5-4800-b57e-379709302934)

dahaca reviewed (approved):
Thank you for refactoring the clipboard util!

matejkriz commented:
@komret passphrase e2e test (touching receive address flow) failing, please double check 
https://gitlab.com/satoshilabs/trezor/trezor-suite/-/jobs/4415494964

bosomt commented:
**QA NOK**

when displaying long xpub on device you can only see first characters of xpub, not sure if its expected.
Problematic xpubs: Cardano ADA and BTC taproot

![image](https://github.com/trezor/trezor-suite/assets/31506317/842b403c-630c-4e69-bc8b-19e76df52c24)

Hannsek commented:
@trezor/qa can you retest this with latest fw prom master? [This](https://github.com/trezor/trezor-firmware/pull/3177) should be the fix.

bosomt commented:
**QA OK**

checked BTC Taproot xpub and Cardano xpubs

**Info:**
 - Suite version: desktop 23.10.0 (fa5f737baa882c2ee12c368da444743e633d040e)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuiteDev/23.10.0 Chrome/112.0.5615.204 Electron/24.5.1 Safari/537.36
 - OS: MacIntel
 - Screen: 1440x900
 - Device: Trezor T2B1 2.6.2 regular (revision 6e5967af25de87bebd331d697e6ef940e64286ef)
 - Transport: BridgeTransport 2.0.32
